### PR TITLE
[OSDOCS-1699]: Documents Autoscaling in OSD

### PR DESCRIPTION
**Pull request by EricPonvelle**
This pull request updates the OSD documentation so that, like the ROSA documentation, the autoscaling concept is documented.

JIRA: https://issues.redhat.com/browse/OSDOCS-1699

Preview: 

**OSD** - https://deploy-preview-35710--osdocs.netlify.app/openshift-dedicated/latest/nodes/nodes-machinepools-autoscaling-about.html

**ROSA** - https://deploy-preview-35710--osdocs.netlify.app/openshift-rosa/latest/nodes/nodes-machinepools-autoscaling-about.html

### CODE REVIEW

**Comment by EricPonvelle** [comment]
@okashi18 PTAL - this is a very early draft, but I managed to successfully reuse the content from ROSA with OSD. I think all of the commands and their output will need updating, but do you see any other areas of concern?

Thanks!

**Comment by jonquilwilliams** [comment]
@EricPonvelle Just FYI - We don't merge or CP to the enterprise branches right now. OSD and ROSA are in the dedictaed-4 branch only.

**Comment by 0kashi** [comment]
For ROSA - LGTM
For OSD - we need to add one note that, for now, autoscaling only applies to clusters purchased through Red Hat Marketplace

**Comment by EricPonvelle** [comment]
@okashi18 I added the note to the OSD side - https://deploy-preview-35710--osdocs.netlify.app/openshift-dedicated/latest/nodes/nodes-machinepools-autoscaling-about.html#machine-autoscaling

Nothing was changed on ROSA.

**Comment by 0kashi** [comment]
OSD: Can we move that note up to the beginning of the "Machine Autoscaling" section?
ROSA: lgtm

**Comment by EricPonvelle** [comment]
@okashi18 Good idea. I moved it to right after the header on OSD. Sending this to review unless you have any other changes.

File: modules/ocm-enabling-autoscaling-nodes.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// nodes/nodes/nodes-enabling-autoscaling-nodes.adoc
+
+[id="ocm-enabling-autoscaling_{context}"]
+= Enabling autoscaling nodes in an existing cluster using OCM
+
+Enable autoscaling for worker nodes in the machine pool definition from the OCM console.
+
+.Procedure
+
+. From the link:https://console.redhat.com/[OCM console], navigate to the *Clusters* page and select the cluster for which you wish to enable autoscaling.
```
Comment (bergerhoffer) [inline modules/ocm-enabling-autoscaling-nodes.adoc:13]: "for which you wish to enable autoscaling" sounds off. Suggest rewording to something like "select the cluster that you want to enable autoscaling for" (it's fine to end in a preposition)

File: modules/ocm-enabling-autoscaling-nodes.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+. On the selected cluster, select the *Machine pools* tab.
+
+. Click the "More Options" at the end of the machine pool with autoscaling and select *Scale*.
+
+. On the "Edit node count" dialog, select the *Enable autoscaling* checkbox.
```
Comment (bergerhoffer) [inline modules/ocm-enabling-autoscaling-nodes.adoc:19]: "Edit node count" in bold

File: modules/ocm-enabling-autoscaling-nodes.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+Enable autoscaling for worker nodes in the machine pool definition from the OCM console.
+
+.Procedure
+
+. From the link:https://console.redhat.com/[OCM console], navigate to the *Clusters* page and select the cluster for which you wish to enable autoscaling.
+
+. On the selected cluster, select the *Machine pools* tab.
+
+. Click the "More Options" at the end of the machine pool with autoscaling and select *Scale*.
```
Comment (bergerhoffer) [inline modules/ocm-enabling-autoscaling-nodes.adoc:17]: Is "More Options" actually shown on the page? If so, it should be bold/asterisks. If not, then it should be reworded another way

File: modules/ocm-enabling-autoscaling-nodes.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+. On the "Edit node count" dialog, select the *Enable autoscaling* checkbox.
+
+. Select *Apply* to save these changes and remove autoscaling from the cluster.
```
Comment (bergerhoffer) [inline modules/ocm-enabling-autoscaling-nodes.adoc:21]: This procedure is for enabling autoscaling. I'm assuming that "remove autoscaling" should be something different

File: modules/ocm-disabling-autoscaling-nodes.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+Disable autoscaling for worker nodes in the machine pool definition from the OCM console.
+
+.Procedure
+
+. From the link:https://console.redhat.com/[OCM console], navigate to the *Clusters* page and select the cluster with autoscaling that must be disabled.
+
+. On the selected cluster, select the *Machine pools* tab.
+
+. Click the "More Options" at the end of the machine pool with autoscaling and select *Scale*.
```
Comment (bergerhoffer) [inline modules/ocm-disabling-autoscaling-nodes.adoc:17]: Same comments as the enabling file for this step and the next

File: nodes/scheduling/nodes-scheduler-node-affinity.adoc [hunk 6]
```diff
@@ -30,4 +30,4 @@ include::modules/nodes-scheduler-node-affinity-example.adoc[leveloffset=+1]
 == Additional resources
 
 For information about changing node labels, see
-xref:../../nodes/nodes/nodes-nodes-working.adoc#nodes-nodes-working-updating_nodes-nodes-working[Understanding how to update labels on nodes].
+xref:../nodes/nodes-nodes-working.adoc#nodes-nodes-working-updating_nodes-nodes-working[Understanding how to update labels on nodes].
```
Comment (bergerhoffer) [inline nodes/scheduling/nodes-scheduler-node-affinity.adoc:33]: I can't see that this file is actually used anywhere. But also I think this would probably 404 like the other files with this change.

File: nodes/scheduling/nodes-scheduler-node-names.adoc [hunk 7]
```diff
@@ -27,5 +27,4 @@ include::modules/nodes-scheduler-node-names-configuring.adoc[leveloffset=+1]
 [id="nodes-scheduler-node-names-addtl-resources_{context}"]
 == Additional resources
 
-For information about changing node labels, see xref:../../nodes/nodes/nodes-nodes-working.adoc#nodes-nodes-working-updating_nodes-nodes-working[Understanding how to update labels on nodes].
-
+For information about changing node labels, see xref:../nodes/nodes-nodes-working.adoc#nodes-nodes-working-updating_nodes-nodes-working[Understanding how to update labels on nodes].
```
Comment (bergerhoffer) [inline nodes/scheduling/nodes-scheduler-node-names.adoc:30]: Same comment as above file. There are more `../nodes/nodes` -> `nodes` in the below files if you can check them out to see if they need to be reverted - I'm not going to point out any more instances

File: monitoring/configuring-the-monitoring-stack.adoc [hunk 8]
```diff
@@ -49,7 +49,7 @@ include::modules/monitoring-moving-monitoring-components-to-different-nodes.adoc
 
 * See xref:../monitoring/configuring-the-monitoring-stack.adoc#preparing-to-configure-the-monitoring-stack[Preparing to configure the monitoring stack] for steps to create monitoring config maps
 * xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[Enabling monitoring for user-defined projects]
-* xref:../nodes/nodes/nodes-nodes-working.adoc#nodes-nodes-working-updating_nodes-nodes-working[Understanding how to update labels on nodes]
+* xref:../nodes/nodes-nodes-working.adoc#nodes-nodes-working-updating_nodes-nodes-working[Understanding how to update labels on nodes]
```
[the remark this answers is not in the document]
Comment (EricPonvelle) [reply monitoring/configuring-the-monitoring-stack.adoc:52]: I believe I need to come back to the OCP repos to fix this.

[the remark this answers is not in the document]
Comment (bergerhoffer) [reply monitoring/configuring-the-monitoring-stack.adoc:52]: @EricPonvelle I don't think any locations of these files changed in the OCP repos. So my guess is that you want to revert these

File: modules/ocm-enabling-autoscaling-nodes.adoc [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+. From the link:https://console.redhat.com/[OCM console], navigate to the *Clusters* page and select the cluster that you want to enable autoscaling for.
+
+. On the selected cluster, select the *Machine pools* tab.
+
+. * Click the Options menu {kebab} at the end of the machine pool that you want to enable autoscaling for and select *Scale*.
+
+. On the *Edit node count* dialog, select the *Enable autoscaling* checkbox.
+
+. Select *Apply* to save these changes and enable autoscaling from the cluster.
```
Comment (bergerhoffer) [inline modules/ocm-enabling-autoscaling-nodes.adoc:21]: should "from" be "for" or something similar?

File: modules/ocm-enabling-autoscaling-nodes.adoc [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+Enable autoscaling for worker nodes in the machine pool definition from the OCM console.
+
+.Procedure
+
+. From the link:https://console.redhat.com/[OCM console], navigate to the *Clusters* page and select the cluster that you want to enable autoscaling for.
+
+. On the selected cluster, select the *Machine pools* tab.
+
+. * Click the Options menu {kebab} at the end of the machine pool that you want to enable autoscaling for and select *Scale*.
```
Comment (bergerhoffer) [inline modules/ocm-enabling-autoscaling-nodes.adoc:17]: Same `*` here

**Review by bergerhoffer — COMMENTED** [review]
Just a few minor comments/typos, then this lgtm!
